Updates description to cover incident and two-way alerting (#1551)
DataDog/integrations-extras: signl4/README.md

You can alert your SIGNL4 team about new Alerts in Datadog. Alerts that get resolved in Datadog will automatically close the alert in SIGNL4. In order to do so you need to configure the following.

1. Navigate to the [Webhooks Integration tile][9].

2. On the **Configuration** tab, go to Webhooks and click **New**.

3. Under **New Webhook**, enter a meaningful `Name` and use the SIGNL4 Webhook `URL` (created above) including your team secret, for example:

    ```
    https://connect.signl4.com/webhook/[team-secret]?ExtIDParam=alertId&ExtStatusParam=alertTransition&ResolvedStatus=Recovered
    ```

    Replace `[team-secret]` with your SIGNL4 team secret here.

    ![SIGNL4 Alerts Webhook][3]

4. Copy-paste the following JSON in the `Payload` text box:

    ```json
	{
		"title": "$EVENT_TITLE",
		"message": "$TEXT_ONLY_MSG",
		"link": "$LINK",
		"priority": "$ALERT_PRIORITY",
		"host": "$HOSTNAME",
		"alertScope": "$ALERT_SCOPE",
		"alertStatus": "$ALERT_STATUS",
		"alertId": "$ALERT_ID",
		"alertTransition": "$ALERT_TRANSITION",
		"X-S4-SourceSystem": "Datadog",
		"date": "$DATE",
		"org": {
			"id": "$ORG_ID",
			"name": "$ORG_NAME"
		},
		"id": "$ID"
	}
    ```

> You can adapt the parameters according to your need but please leave alertId, alertTransition and X-S4-SourceSystem unchanged.

Reviewer: ```suggestion
You can adapt the parameters according to your needs, but leave `alertId`, `alertTransition`, and `X-S4-SourceSystem` unchanged.
```

> 5. Click on **Save** to complete the webhook creation.

Reviewer: ```suggestion
5. Click **Save** to create the webhook.
```

For more details, see [Mobile alerting with tracking & escalation for Datadog][7].

> That's it and now you can use your webhook as a notification channel in your Monitor. You can choose @webhook-SIGNL4 here if SIGNL4 is the name of your webhook. When the conditions of the monitor apply your team will receive a new SIGNL4 alert.

Reviewer: ```suggestion
You can now use your webhook as a notification channel in your monitor. Assuming the name of your webhook is SIGNL4, send notifications using `@webhook-SIGNL4`. When the conditions of the monitor apply, your team receives a new SIGNL4 alert.
```

### Datadog Incidents

> You can alert your SIGNL4 team about new Incidents in Datadog. Incidents that get resolved in Datadog will automatically close the alert in SIGNL4. In order to do so you need to configure the following.

Reviewer: ```suggestion
You can alert your SIGNL4 team about new Incidents in Datadog. Incidents that get resolved in Datadog automatically close the alert in SIGNL4. In order to do so, configure the following:
```

1. Navigate to the [Webhooks Integration tile][9].

2. On the **Configuration** tab, go to Webhooks and click **New**.

3. Under **New Webhook**, enter a meaningful `Name` and use the SIGNL4 Webhook `URL` (created above) including your team secret, for example:

    ```
    https://connect.signl4.com/webhook/[team-secret]?ExtIDParam=s4ExternalId&ExtStatusParam=incidentStatus&ResolvedStatus=resolved
    ```

>     Replace `[team-secret]` with your SIGNL4 team secret here.

Reviewer: ```suggestion
    Replace `[team-secret]` in the URL with your SIGNL4 team secret.
```

Author: Thanks @yzhan289. Suggestion approved and I hope all is fine now.

	}
    ```

> You can adapt the parameters according to your need but please leave incidentStatus, s4ExternalId and X-S4-SourceSystem unchanged.

Reviewer: ```suggestion
Adapt the parameters according to your needs, but leave `incidentStatus`, `s4ExternalId` and `X-S4-SourceSystem` unchanged.
```

> 5. Click on **Save** to complete the webhook creation.

Reviewer: ```suggestion
5. Click **Save** to create the webhook.
```

For more details, see [Mobile alerting with tracking & escalation for Datadog][7].

### Incidents Rule

> Now you can create a rule in Datadog under Monitors -> Settings (Incidents) Rules. In the you specify the criteria, i.e. the severity, service, etc. For the "Other attributes" we recommend to use "state:active" that will trigger a new alert and "state:resolved" that will close the alert. Under "Notify" you select your SIGNL4 incident webhook as created above and that's it.

Reviewer: ```suggestion
You can create a rule in Datadog under Monitors -> Settings (Incidents) Rules. In the rule, you specify the criteria, for example, the severity, service, etc. For the "Other attributes", it is recommended to use "state:active" to trigger a new alert and "state:resolved" to close the alert. Under "Notify", select your SIGNL4 incident webhook as created above.
```

![SIGNL4 Incidents Rule][5]

> If you create a new incident now, your SIGNL4 team will receive an alert. If you acknowledge / close the alert in the SIGNL4 app, the incident status will be set to Stable / Resolved.

Reviewer: ```suggestion
If you create a new incident, your SIGNL4 team receives an alert. If you acknowledge or close the alert in the SIGNL4 app, the incident status is be set to Stable or Resolved.
```

> Also, if you set the incident status to Resolved in Datadog, this will close the alert in SIGNL4.

Reviewer: ```suggestion
Also, if you set the incident status to Resolved in Datadog, this closes the alert in SIGNL4.
```

> ### Status Updates for Datadog Incidents

Reviewer: ```suggestion
### Status updates for Datadog incidents
```

> For Datadog incidents you can update the status to Stable / Resolved directly from your SIGNL4 app by acknowledging / closing the respective alert.

Reviewer: ```suggestion
For Datadog incidents, you can update the status to Stable or Resolved directly from your SIGNL4 app by acknowledging or closing the respective alert.
```

In order to configure this back channel, go to your SIGNL4 web portal and then to Teams -> Apps. Search for the Datadog connector app and click "Create" there. You can find more information directly within the app settings.

![Datadog Connector App][6]

> You need to configure the following.

Reviewer: ```suggestion
You need to configure the following:
```

> Datadog URL: The URL of your Datadog instance, e.g. https://app.datadoghq.com/ or https://app.datadoghq.eu/.  
> Datadog API Key: Your Datadog API key. You can find or create this one in Datadog under your account, Organization Settings -> API Keys.  
> Datadog Application Key: Your Datadog application key. You can find or create this one in Datadog under your account, Organization Settings -> Application Keys.  
> Acknowledgement as Stable: Acknowledgements set the incident status to Stable.  

Reviewer: ```suggestion
- Datadog URL: The URL of your Datadog instance, for example, https://app.datadoghq.com/ or https://app.datadoghq.eu/.  
- Datadog API Key: Your Datadog API key. You can find or create an API key in Datadog under your account, Organization Settings -> API Keys.  
- Datadog Application Key: Your Datadog application key. You can find or create an application key in Datadog under your account, Organization Settings -> Application Keys.  
- Acknowledgement as Stable: Acknowledgements set the incident status to Stable.  
```

The SIGNL4 integration does not include any service checks.

## Troubleshooting
Need help? Contact [SIGNL4 Support][8].

[1]: https://www.signl4.com
[2]: https://raw.githubusercontent.com/DataDog/integrations-extras/master/signl4/images/signl4-phone.png
[3]: https://raw.githubusercontent.com/DataDog/integrations-extras/master/signl4/images/datadog-alerts-webhook.png
[4]: https://raw.githubusercontent.com/DataDog/integrations-extras/master/signl4/images/datadog-incidents-webhook.png
[5]: https://raw.githubusercontent.com/DataDog/integrations-extras/master/signl4/images/datadog-incidents-rule.png

> [6]: https://raw.githubusercontent.com/DataDog/integrations-extras/master/signl4/images/signl4-datadog-connector-app.png

Reviewer: These links 404 for me.

Author: Thanks a lot for your comments and changes @urseberry. The missing files are here:
https://github.com/rons4/integrations-extras/tree/master/signl4/images
It seems I cannot get them into this PR. Probably my fault. Should I create a separate PR?

Reviewer: Yes, if you are having trouble getting the images into this PR, you can create a separate one.

Reviewer 2: If you are including them in a separate PR, can you get rid of these links for now? The validation is failing because these files don't exist.
